test/distributed/fsdp/test_fsdp_comm_hooks.py

```diff
@@ -73,46 +74,89 @@ def forward(self, x):
 class DummyState(object):
 
     __slots__ = [
-        "process_group"
+        "process_group",
+        "noise"
     ]
 
-    def __init__(self, process_group):
+    def __init__(self, process_group, noise):
         self.process_group = process_group
+        self.noise = noise
 
 class DummyHook(object):
 
-    def dummy_hook(self, state: DummyState, grad: torch.Tensor):
+    def dummy_hook_for_no_shard_fsdp(self, state: DummyState, grad: torch.Tensor):
+        """
+        This communication hook is for illustration and testing purpouse only.
+        This communication hook is used during FSDF ``NO_SHARD`` training. It adds some moise to
+        the provided ``grad`` porameter and uses ``all_reduce`` to communicate full, flattened,
+        unsharded gradient.
+        """
+        grad.add_(state.noise)
+        dist.all_reduce(grad, group=state.process_group)
+
+    def custom_reduce_scatter(self, output, input, group=None):
+        """
+        This function is for illistrative purpose only.
+        It is meant to implement a custom reduce and scatter
+        of a flattened tensor to all processes in a group.
+        Currently a no-op.
+        """
         pass
 
+    def dummy_hook_for_sharded_fsdp(self, state: DummyState, grad: torch.Tensor, output: torch.Tensor):
+        """
+        This communication hook is for illustration and testing purpouse only.
+        This communication hook is used during FSDF ``FULL_SHARD`` or ``SHARD_GRAD_OP`` training.
+        It adds some moise to the provided ``grad`` porameter and uses
+        ``reduce_scatter`` for gradient communication and stores a sharded gradient in ``output``.
+        """
+        grad.add_(state.noise)
+        self.custom_reduce_scatter(
+            output, grad, group=state.process_group
+        )
+
 class TestCommunicationHooks(FSDPTest):
 
     @skip_if_lt_x_gpu(2)
     @parametrize(
         "sharding_strategy",
         [
-            ShardingStrategy.NO_SHARD
+            ShardingStrategy.NO_SHARD,
+            ShardingStrategy.FULL_SHARD,
+            ShardingStrategy.SHARD_GRAD_OP
         ])
     def test_default_communication_hook_behavior(
         self,
         sharding_strategy: Optional[ShardingStrategy]
     ):
         """
         Tests FSDP's default communication hook's behavior and correctness.
+        This test creates a simple net ``1 X N``, where ``N`` - is the number of workers.
```

Comment:
nit: Something like this?
```suggestion
        This test creates a simple linear net with weight shape ``1 X N``, where ``N`` - is the number of workers.
```